### ISSUE TITLE
Theme: Broken automatic update (404 error)

### ISSUE THREAD

**Issue by TopheC** [qooob/authentic-theme#787]
Hello,

* What is your server-side OS and its version? : Ubuntu 14.04.5 LTS 
* What is your client-side OS, browser and its versions? Linux Mint 18.1 Serena, Firefox 53.0.2 (64bits)
* What are the steps to reproduce the issue :

I was unable to update, as usually, from previous version 18.48 of the theme. My browser fired a 404 from github.
Same problem of bad url from the download link on readme.md !

I saw a "-2" on the next version name in my Webmin Dashboard panel, so i try to install the new theme version from the "install theme" tab with this url : https://github.com/qooob/authentic-theme/releases/download/18.49/authentic-theme-18.49-2.wbt.gz
I just add "-2" to the name and before extension.

Maybe help some folks :-)

**Comment by jvieille** [comment]
Something is wrong with this update.
 Latest available release 18.49-2 does not proceed

**Comment by qooob** [comment]
I'm sorry, guys, my bad. I didn't add `18.49` as a release package but only ``18.49-2``. Now it's going to work. Please try.

**Comment by jvieille** [comment]
Same:
- update runs and does nothing
- Download returns 404 
https://github.com/qooob/authentic-theme/releases/download/18.49-2/authentic-theme-18.49-2.wbt.gz

**Comment by qooob** [comment]
Oohh. Yeah. The only way to overcome this right now is to go to theme settings and force update the  theme. 

I was going to enroll this feature for a long time but it happened to be back-incompatible.

In case Git command is installed force update button will appear in theme settings and will let you update very easily.

**Comment by jvieille** [comment]
How do I install Git command?
thanks

**Comment by qooob** [comment]
I think `apt-get install git` or using Webmin packages module.

**Comment by Infowan** [comment]
TopheC's method solved my problem. All servers up to date. Thak you TopheC!

**Comment by jvieille** [comment]
TopheC' method was simpler.
thanks

**Comment by TopheC** [comment]
@jvieille and @Infowan, glad it helps you :+1: 
@qooob , maybe some new lines on the "Troubleshoot" section of the readme.md, if you please ?

**Comment by qooob** [comment]
Thank you, but I don't think it's necessary as this will be resolved very soon with next Webmin/Usermin release. It's only applicable to the users that missed 18.49 update.

Besides, there is a section explaining how to update the theme manually, using console script. :wink:

**Comment by TopheC** [comment]
Ok, it's my ITIL Knowledge Management side : a bug > a known error > a solution > known errors database (aka FAQ) updated with the solution > available publicly... you're the boss :wink:

**Comment by Francewhoa** [comment]
We are confirming this bug. It is recurring every few release. Each time that bug happens users are not able to update using the interface.

### Steps to reproduce bug
1. Using:
      1. Operating system 	Debian Linux 8
      1. Webmin version 	1.840
      1. Virtualmin version 	5.07
      1. Theme version 	Authentic Theme 18.48
1. Go to Virtualmin dashboard page, under "Theme version" row, click on [that "Update" button](https://cloud.githubusercontent.com/assets/119942/26378706/94c5fb02-3fca-11e7-8aa8-14954ad8c9f2.jpg).
1. [That red progress bar](https://cloud.githubusercontent.com/assets/119942/26378736/b74521bc-3fca-11e7-88cb-caf6b4dac6bf.jpg) will be display at the top of the page


1. The user is automatically redirected to Virtualmin dashboard. The updated failed. The version is still Authentic Theme 18.48. Expected result is the theme should automatically be updated and a overlay window should be display titled "Update Notice".

### Screenshots
![dashboard---screenshot---466](https://cloud.githubusercontent.com/assets/119942/26378706/94c5fb02-3fca-11e7-8aa8-14954ad8c9f2.jpg)
 
![progress---screenshot---467](https://cloud.githubusercontent.com/assets/119942/26378736/b74521bc-3fca-11e7-88cb-caf6b4dac6bf.jpg)

**Comment by qooob** [comment]
Yes, it's already fixed in 18.49+. Update manually or forces-update from theme's settings.

**Comment by Francewhoa** [comment]
For those neither using command line nor Git. Here is an easy and quick fix that worked for us. Using only the Virtualmin interface simply clear the theme cache, then try to update again. It usually works. Details below.

## Temporary fix workaround
Steps
1. Using Virtualmin dashboard page, under "Theme version" row, click on "Options" button
1. The next page is titled "Webmin Themes". Scroll all the way down that page. Click on "Clear cache" button.
1. Go back to Virtualmin dashboard page, under "Theme version" row, click on "Update" green button
1. Wait. A red progress bar will be display at the top of the page.
1. When done an overlay window will open. Titled "Update Notice". Done you have successfully updated the theme.

**Comment by Francewhoa** [comment]
@qooob :) Thanks for your reply

Here is a suggested improvement. If not already done, on Virtualmin dashboard, when the user click on the theme "Update" button, before starting the update process, we suggest to automatically clear the theme cache, then proceed with the theme update process. That would reduce risk of future broken automated updates. 

This is related to my comment above at https://github.com/qooob/authentic-theme/issues/787#issuecomment-303549882

**Comment by rstiger** [comment]
Yep I'm having the same problem updating theme I'm getting 404 error.
I tried clearing the cache as suggested by Francewhoa but no luck.
Operating system	Ubuntu Linux 14.04.4
Webmin version	1.840 
Virtualmin version	5.07
Theme version	Authentic Theme 18.48. Latest available release 18.49-5    Update Changelog Download Options
Time on system	Tuesday, May 23, 2017 7:49 PM
Kernel and CPU	Linux 3.13.0-119-generic on x86_64
Processor information	Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, 4 cores

I guess I will either install manually or just wait see...

**Comment by rstiger** [comment]
ok so I waited long enough and just manually installed and we are now up to date.

I went ahead and download the update from-
https://github.com/qooob/authentic-theme/releases/download/18.49/authentic-theme-18.49-5.wbt.gz

And then installed theme from upload. 

Operating system	Ubuntu Linux 14.04.4
Webmin version	1.840 
Usermin version	1.710
Virtualmin version	5.07
Theme version	Authentic Theme 18.49-5
Time on system	Tuesday, May 23, 2017 7:59 PM
Kernel and CPU	Linux 3.13.0-119-generic on x86_64
Processor information	Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, 4 cores
System uptime	1 hours, 24 minutes
Running processes	214
CPU load averages	0.00 (1 min) 0.01 (5 mins) 0.05 (15 mins)
Real memory	3.77 GB total / 1.45 GB used
Virtual memory	7.81 GB total / 0 bytes used
Local disk space	241 GB total / 195.25 GB free / 45.74 GB used
Package updates	All installed packages are up to date

I'm up to date now but others using the "Update" button might have same previous issues till it's fixed

**Comment by mat-l** [comment]
Clear cache didn't work for me, but the manual upload was working fine here 👍

**Comment by Francewhoa** [comment]
We're confirming the "Force update theme" button is now display with the latest Theme version Authentic Theme 18.49-**6**
Yay :)
![force-update---screenshot---ubertus---018](https://cloud.githubusercontent.com/assets/119942/26467197/6abcbb52-4146-11e7-93c3-acfbf0aeb640.png)

**Comment by Vitusware** [comment]
I don't have a "Force update theme" button..

I still have Authentic Theme 18.48, 
whatever I try,  I keep getting the 404 error.. "this is not the page you are looking for" :( 

Is there another link I can use to update? and how to update manually?

**Comment by johnnyaston** [comment]
I have the "Force update theme" button but it still doesn't work. I would run a manual update but I'm having problems with this also.

**Comment by qooob** [comment]
From 18.48 you would need to run second option, which still will update you with the release.

To see this button you need to install Git command.

**Comment by Vitusware** [comment]
Total noob here :D 

so how do I do that ? ( install Git command.)

**Comment by johnnyaston** [comment]
Thank qooob, yes I have Git command and the button is showing. I tried the second option and it reports that the update was successful - but it turns out not to be the case.

**Comment by qooob** [comment]
To update the theme using package and get `update-script.sh` pulled to the latest use the package updates.

Go to _Webmin/Webmin Configuration/Themes/Install Theme/From ftp or http URL_, check the radio-button, paste the following link and hit _**Install Theme**_

`http://github.com/qooob/authentic-theme/releases/download/18.49/authentic-theme-18.49.wbt.gz`

**Comment by johnnyaston** [comment]
Great thanks qooob, that worked for me! All of the previous links mentioned were doing a 404.

**Comment by qooob** [comment]
Yes, there was back-incompatible change. Sorry.

**Comment by Vitusware** [comment]
Thank you qooob, that worked perfectly, good support  thank you!
![image](https://cloud.githubusercontent.com/assets/29035591/26545348/8c45bc70-4465-11e7-83e9-ee0f3de8adfd.png)

**Comment by pieter-groeneweg** [comment]
I have a button to force install.. I am on 18.48.
After clicking the force install, The announced version update changed from 18.49-1 to 18.49-7... 
And I am still on 18.48

**Comment by qooob** [comment]
Follow these instructions
https://github.com/qooob/authentic-theme/issues/787#issuecomment-304617121

**Comment by pieter-groeneweg** [comment]
That works. Thanks
